docs/adr/0007_resource_download.md

```diff
@@ -0,0 +1,194 @@
+* Ensuring that transformers can be registered dynamically (via the plugin manager) or inbuilt (via the OCM CLI)
+* Reusing the existing binding libraries to interact with OCM repositories (by interacting only with `blob.ReadOnlyBlob`)
+
+**Important Note**: The `BlobTransformer` does *not* work on the Resource/Source level. It is intentionally independent of the resource description or access method. Instead, the blob’s MIME-type or other self-declared metadata is used to guide transformation logic. This is because it is expected that the transformation should work on any data stream independent of resources or sources.
+
+---
+
+## Decision Drivers
+
+* **Simplicity of Architecture**: The model avoids over-engineering and keeps resource transformation localized and decoupled.
+* **Extensibility**: The plugin-based approach allows for new transformers to be added without changing the core CLI logic.
+* **Maintainability**: Clear interfaces and a separation of concerns will reduce technical debt and allow for more robust testing and evolution of the system.
+
+---
+
+## Outcome
+
+We will implement a plugin-driven, extensible transformation system based on the `BlobTransformer` contract that supports interpreting blobs in the CLI, without requiring custom user code.
+
+---
+
+## An Easy-to-Use `bindings/go/blob/transformer` Library
+
+The core of the proposed system is a new module:
+
+```text
+bindings/go/blob/transformer
+```
+
+This module is responsible for registering and invoking `BlobTransformer` implementations, whether inbuilt or provided by plugins. It acts as a central point for managing transformation logic and supporting configuration resolution.
+
+---
+
+## `BlobTransformers` and Contracts
+
+The key interface provided by this module is:
+
+```go
+package transformer
+
+import (
+    "context"
+
+    "ocm.software/open-component-model/bindings/go/blob"
+    "ocm.software/open-component-model/bindings/go/runtime"
+)
+
+type BlobTransformer interface {
+    // TransformBlob transforms the given blob data according to the specified configuration.
+    // It returns the transformed data as a blob.ReadOnlyBlob or an error if the transformation fails.
+    TransformBlob(ctx context.Context, input blob.ReadOnlyBlob, config runtime.Typed) (blob.ReadOnlyBlob, error)
+}
+```
+
+This allows flexible input and output handling based solely on `blob.ReadOnlyBlob`, enabling powerful chaining and reuse.
+
+---
+
+### Example Configuration
+
+The transformer system will look for a `transformer.blob.ocm.software/v1alpha1` config in the `.ocmconfig` file:
+
+```yaml
+type: transformer.blob.ocm.software/v1alpha1
+transformers:
+- id: extract-oci-artifact
+  spec:
+    type: extract.oci.artifact.ocm.software/v1alpha1
+    # additional configuration for the transformer
+```
+
+This allows transformer registration via configuration, enabling portable setups and declarative ocm configuration structures.
```

Comment:
I'm struggling with this part:
> This allows transformer registration via configuration

How? I mean this is just passed to the transformer, right? That means it already "exists" technically. This allows to configure that for sure. But I don't see how this allows "registration". Like, for me, registration would mean that for example, if it's a binary, it would point to the binary to load during startup or something.

I guess my question is, what part of this is allowing registration? The fact that the config type `extract.oci.artifact.ocm.software/v1alpha1` exists in the config? Is that enough?

Reply:
For me the registration is coming from the `id`. The id defines an instance of the config that you can directly access (see the config for that)

Reply:
Ahhha. Okay. I thought so. That makes sense. Can we mention that please to make it explicit? 🙇